LegacyAutoloader.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+class LegacyAutoloader
+{
+    public function __construct()
+    {
+        spl_autoload_register(array($this, 'load_class'));
+    }
+
+    public static function register()
+    {
+        new LegacyAutoloader();
+    }
+
+    public function load_class($className)
+    {
+        if (strpos($className, 'Piwik\\') === 0) {
+            $newName = 'Matomo' . substr($className, 5);
+            if (class_exists($newName)) {
+                class_alias($newName, $className);
+            }
+        }
+    }
+}
+
+LegacyAutoloader::register();
```

Comment:
Maybe I missed it but could we add maybe a few basic tests that this works? 

In general behaviour seems otherwise good: https://3v4l.org/TAS8d

Can we assume, that if a file containing a Matomo class was loaded manually through `include/require`, then the logic would not work? (Probably not an issue but something to be aware of eg if we wanted to do that for all classes in core).

Reply:
No, that works as well. Added some simple tests for it. See https://github.com/matomo-org/matomo/pull/15253/commits/de2fea66381aa5d5d2631bf535da68cd2b42d5cb